File: ui/src/main/java/io/temporal/samples/moneytransfer/model/TransferStatus.java
package io.temporal.samples.moneytransfer.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class TransferStatus {
    private int progressPercentage;
    private String transferState;
    private String workflowStatus;
    private ChargeResponse chargeResult;
    private int approvalTime;
}

File: dotnet/src/model/TransferStatus.cs
using System.Text.Json.Serialization;

namespace MoneyTransfer;

public class TransferStatus
{
    [JsonPropertyName("progressPercentage")]
    public int ProgressPercentage { get; set; }
    [JsonPropertyName("transferState")]
    public string TransferState { get; set; }
    [JsonPropertyName("workflowStatus")]
    public string WorkflowStatus { get; set; }
    [JsonPropertyName("chargeResult")]
    public DepositResponse DepositResponse { get; set; }
    [JsonPropertyName("approvalTime")]
    public int ApprovalTime { get; set; }

    public TransferStatus()
    {
        ProgressPercentage = 0;
        TransferState = "";
        WorkflowStatus = "";
        DepositResponse = new DepositResponse();
        ApprovalTime = 0;
    }

    public TransferStatus(int progressPercentage, string transferState, string workflowStatus, DepositResponse depositResponse, int approvalTime)
    {
        ProgressPercentage = progressPercentage;
        TransferState = transferState;
        WorkflowStatus = workflowStatus;
        DepositResponse = depositResponse;
        ApprovalTime = approvalTime;
    }
}
